### ISSUE TITLE
Java-tests fail when using annotated test-methods

### ISSUE THREAD

**Issue by jarmoni** [radekg/kafka-cluster-tools#11]
When modifying `SimpleTypesTest` by using the _JUnit4-approach_ (test-method annotated with `@Test`) the consumer never receives a result:
```
java.lang.AssertionError: Eventually executed 99 times and failed 99 times after 10000 milliseconds.

	at org.junit.Assert.fail(Assert.java:88)
	at com.gruchalski.kafka.test.SimpleTypesTest$1.accept(SimpleTypesTest.java:96)
	at com.gruchalski.kafka.test.SimpleTypesTest$1.accept(SimpleTypesTest.java:44)
	at java.util.Optional.ifPresent(Optional.java:159)
	at com.gruchalski.kafka.test.SimpleTypesTest.testClusterSetup(SimpleTypesTest.java:44)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:51)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:237)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
```

**Comment by jarmoni** [comment]
Resolved the following:
* Entry is successfully consumed but thread  in `AsyncUtil` runs until timeout
* The `assertEquals` in line 89 and 90 have no effect. Even when using _old_ JUnit3-style by extending `TestCase`. The test is green in this case even if I replace:
```
assertEquals(consumed.get().value(), aValue);
```
with
```
assertEquals(consumed.get().value(), "xyz");
```
This is because `assert` is executed in different thread.

**Comment by jarmoni** [comment]
Should be fixed in pull-request #12
